src/core/processing/qgsprocessingutils.cpp

```diff
@@ -443,14 +444,27 @@ QgsFeatureSink *QgsProcessingUtils::createFeatureSink( QString &destination, Qgs
     QString layerName;
     QString format;
     bool useWriter = false;
-    parseDestinationString( destination, providerKey, uri, layerName, format, options, useWriter );
+    parseDestinationString( destination, providerKey, uri, layerName, format, adaptedCreateOptions, useWriter );
+
+    QgsFields newFields = fields;
+    // If the result is a geopackage layer and there is already a field name FID requested which
+    // might contain duplicates, make sure to generate a new field with a unique name instead
+    // that will be filled by ogr with unique values.
+    if ( options.testFlag( QgsProcessingUtils::FidMayContainDuplicates ) && format == QLatin1String( "GPKG" ) && !adaptedCreateOptions.contains( QStringLiteral( "FID" ) ) )
+    {
+      int fidIdx = fields.lookupField( QStringLiteral( "FID" ) );
```

Comment:
Ok, moving on with the discussion :)

Am I missing something here, or won't the attributes added to this sink contain the existing fid value, and accordingly be one row out of sync with the actual sink fields?

Reply:
Correct. As soon as it's run with more fields it gets out of sync.

Will have to think of something else... But that's gonna be 3.4.1